solo/hmac_secret.py

```diff
@@ -86,7 +87,15 @@ def simple_secret(
     client.origin = f"https://{client.host}"
     client.user_id = user_id
     # user = {"id": user_id, "name": "A. User"}
-    credential_id = binascii.a2b_hex(credential_id)
+
+    #check for even number of hex characters (case-insensitive)
+    pattern = re.compile("^([A-Fa-f0-9]{2})+$")
```

Comment:
Regexes... :cold_sweat:

Reply:
what's bad about regex? I mean you have a better idea for checking whether something is proper hex without everything breaking down?

Reply:
see below